### PR TITLE
Libxml 2 9 1

### PR DESCRIPTION
**Pull request by mwieder**
Updated libxml library to the latest version (2.9.1). This is a necessary step in the process of providing xslt functionality because the libxslt library requires at least libxml 2.6.x.

### CODE REVIEW

File: libxml/Makefile [hunk 1]
```diff
@@ -3,14 +3,15 @@ TYPE=archive
...
+SOURCES= buf.c c14n.c catalog.c chvalid.c debugXML.c dict.c DOCBparser.c encoding.c \
 	entities.c error.c globals.c hash.c HTMLparser.c HTMLtree.c legacy.c list.c \
+	nanoftp.c nanohttp.c \
 	parser.c parserInternals.c pattern.c relaxng.c SAX.c schematron.c threads.c \
 	tree.c uri.c valid.c xinclude.c xlink.c xmlIO.c xmlmemory.c xmlmodule.c \
 	xmlreader.c xmlregexp.c xmlsave.c xmlschemas.c xmlschemastypes.c xmlstring.c \
 	xmlunicode.c xmlwriter.c xpath.c xpointer.c SAX2.c
 
-CUSTOM_DEFINES=
+CUSTOM_DEFINES=LIBXML_XPATH_ENABLED
```
Comment (runrevmark) [inline libxml/Makefile]: I don't think this define is needed here, since we put it in xmlversion.h.

**Comment by runrevmark** [comment]
Hi Mark, thanks for this - just a couple of things to check / attend to as mentioned in the comments above. Mark.

**Comment by mwieder** [comment]
Thanks. Good catches. I'm so glad not to have to change the libxml files themselves - it's enough of a pain having to restructure them into the src and include folders to accommodate the makefiles. I reverted the changes, built, and tested and it seems to be working fine with the new library. This paves the way for the xslt functions.

**Comment by runrevmark** [comment]
Hi Mark,

I've pulled this in via the branch livecode-thirdparty/feature-libxml_2_9_1 and updated the projects / Makefiles for the other platforms. I've also updated the submodule pointer in livecode/develop and there are a couple of tweaks to revXML on Windows to make things compile.

I did have to make minor modifications to some of the config files to make sure libxml-2.9.1 compiles on all the platforms, but nothing major.

Thanks for your work,

Mark.

**Pull request closed without merging**